src/SDKs/Batch/DataPlane/Azure.Batch.Unit.Tests/BatchSharedKeyCredentialUnitTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Azure.Batch.Unit.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Batch.Protocol;
+    using Microsoft.Net.Http.Server;
+    using Xunit;
+
+    public class BatchSharedKeyCredentialUnitTests
+    {
+        private const string url = "http://localhost:2055";
+
+        [Theory]
+        [MemberData(nameof(HttpMethods))]
+        public async Task TestSharedKeySigning_ContentLength(HttpMethod httpMethod)
```

Comment:
Test name does not describe what is being asserted/tested.  Reading the code this one seems to contain a number of assertions, not all of which relate to content length; and the content length assertions vary from HTTP method to HTTP method.  So I'm not sure what to suggest as a better name except that it should be a statement which can be true or false.  (You might want to consider breaking out the different HTTP methods so you could have more specific test names for each, e.g. `HttpDelete_WithSharedKeySigning_DoesNotIncludeContentLength`.)